src/d/d_camera.cpp

```diff
@@ -2008,10 +2093,308 @@ cSAngle dCamera_c::forwardCheckAngle() {
+                        case 3:
+                        case 4:
+                        case 7:
+                        case 8:
+                        case 13:
+                        case 14:
+                        case 15:
+                            wall_up_distance = 150.0f;
+                            break;
+                      
+                        default:
+                            wall_up_distance = 110.0f;
+                            break;
+                      }
+                }
+                else if (proc_name == PROC_NPC_MD) {
+                    wall_up_distance = 130.0f;
+                }
+                else if (proc_name == PROC_Obj_Try) {
+                    wall_up_distance = 200.0f;
+                }
+                else {
+                    wall_up_distance = 110.0f;
+                }
+            }
+        }
+    }
+
+    cXyz eye = m050;
+    cSGlobe direction = m03C;
+
+    if (chkFlag(0x2000) && mpLockonTarget) {
+        f32 sight_radius = radiusActorInSight(mpPlayerActor, mpLockonTarget);
+        if (sight_radius > 0.0f) {
+            if (sight_radius >= 3500.0f) {
+                sight_radius = 3500.0f;
+            }
+            m14C += (sight_radius - m14C) * 0.33f;
+            res |= 0x40;
+        }
+        else {
+            m14C -= m14C * 0.08f;
+        }
+
+        f32 fVar15 = 1.0f;
+        if (m108 < 10) {
+            fVar15 = m108 / 10.0f;
+        }
+        
+        direction.R(m14C * fVar15 + direction.R());
+        eye = m044 + direction.Xyz();
+    }
+
+    if ((i_flags & 0x40) && m364 != 0) {
+        cSGlobe cStack_3e4 = m36C - m044;
+        if (direction.V() < cStack_3e4.V()) {
+            cSAngle local_408 = mDirection.V();
+            local_408 += (cStack_3e4.V() - local_408) * 0.05f;
+            direction.U(local_408);
+            eye = m044 + direction.Xyz();
+            res |= 0x20;
+        }
+    }
+
+    dBgS_CamLinChk_NorWtr lin_chk1;
+    dBgS_CamLinChk_NorWtr lin_chk2;
+
+    float fVar2;
+
+    cXyz local_2fc; /* 0x2FC */
+    cXyz mid; /* 0x2F0 */
+    cXyz local_2e4; /* 0x2E4 */
+    cXyz cross_prod; /* 0x2D8 */
+    cXyz cross1; /* 0x2CC */
+    cXyz cross2; /* 0x2C0 */
+    cXyz local_2b4; /* 0x2B4 */
+    cXyz local_2a8; /* 0x2A8 */
+    cXyz local_29c; /* 0x29C */
+    cXyz local_290; /* 0x290 */
+    cXyz local_284; /* 0x284 */
+    cXyz local_278; /* 0x278 */
+
+    if (lineBGCheck(&mCenter, &eye, &lin_chk1, i_flags)) {
+        cM3dGPla* plane1 = dComIfG_Bgsp()->GetTriPla(lin_chk1);
+        cM3dGPla* plane2 = NULL;
+        if ((i_flags & 0x20) == 0) {
+            curr_hit_type = 2;
+        }
+        else {
+            if (lineBGCheck(&eye, &mCenter, &lin_chk2, i_flags)) {
+                plane2 = dComIfG_Bgsp()->GetTriPla(lin_chk2);
+                float dot_prod = PSVECDotProduct(plane1->GetNP(), plane2->GetNP());
+                PSVECCrossProduct(plane1->GetNP(), plane2->GetNP(), &cross_prod);
```

Comment:
```suggestion
                float dot_prod = VECDotProduct(plane1->GetNP(), plane2->GetNP());
                VECCrossProduct(plane1->GetNP(), plane2->GetNP(), &cross_prod);
```
These defines get replaced with either the `PS` or `C_` functions depending on whether the game is compiled in release or debug mode, so the functions shouldn't be called directly.